expo/expo
# [firebase-analytics] Add Pure JS Analytics module for Expo Client

# Why

This PR adds a Pure JS Firebase Analytics module to `expo-firebase-analytics`, which is used on the standard Expo Client to log events. Using Firebase Analytics is otherwise not possible as the firebase JS SDK Analytics module requires browser features, and the native Firebase SDK requires a google-services file to be bundled with the app, and cannot be  (re-)initialized correctly.

The pure JS implementation makes it possible for users to use and test firebase analytics without having to create a standalone build or use a custom client.

# How

A Pure JS Firebase Analytics implementation was created which uses the Google Measurement Protocol version 2. This protocol is used by the gtag.js/firebase-sdk itself and communicates with the `https://www.google-analytics.com` domain. This protocol is not officially documented by Google yet, so it may be affected by API changes in the future.
In order to log events to the Google Analytics server, a `measurementId` is needed. This can be configured in `app.json`, [through the new `web.config.firebase` key.](#7086) 

# Test Plan

Some manual steps need to be performed in order to test this using the test-suite.
I've been working on getting google-services config dynamically linked in test-suite using `app.config.js` but that is not ready yet and requires updates to expo-cli.

- Follow the steps [to setup "native" firebase with the test-suite](#6662)
- Make sure to include the web-configuration
- Open the project on the standard Expo Client
- Run the "Firebase Analytics" tests
- After a short while (couple mins at most) the Analytics events show up in your firebase project
